Links in aggregate-third-party-report.html point to third-party-report.html

When creating an aggregated report with `license:aggregate-third-party-report`, the resulting file contains links to the (non existing) file `third-party-report.html`

Either fix the name of the file or use relative links (i.e `#Overview` instead of `third-party-report.html#Overview`).

The fix is probably just changing one line in `license-maven-plugin\src\main\java\org\codehaus\mojo\license\ThirdPartyReportRenderer.java`: 

        sink.link( "./third-party-report.html#" + getText( "report.overview.title" ) );

to 

        sink.link( "#" + getText( "report.overview.title" ) );

I checked out the project and could compile it ./mvnw clean install but I can't run it with Maven 3.6:

    Caused by: java.lang.ClassCastException: java.lang.String cannot be cast to org.apache.maven.artifact.repository.ArtifactRepository
        at org.apache.maven.project.artifact.DefaultMavenMetadataCache.repositoriesHashCode (DefaultMavenMetadataCache.java:155)
        at org.apache.maven.project.artifact.DefaultMavenMetadataCache.access$100 (DefaultMavenMetadataCache.java:43)
        at org.apache.maven.project.artifact.DefaultMavenMetadataCache$CacheKey.<init> (DefaultMavenMetadataCache.java:80)
        at org.apache.maven.project.artifact.DefaultMavenMetadataCache.newCacheKey (DefaultMavenMetadataCache.java:340)
        at org.apache.maven.project.artifact.DefaultMavenMetadataCache.get (DefaultMavenMetadataCache.java:307)
        at org.apache.maven.project.artifact.MavenMetadataSource.retrieve (MavenMetadataSource.java:157)
        at org.apache.maven.repository.legacy.resolver.DefaultLegacyArtifactCollector.recurse (DefaultLegacyArtifactCollector.java:551)
        at org.apache.maven.repository.legacy.resolver.DefaultLegacyArtifactCollector.collect (DefaultLegacyArtifactCollector.java:148)
        at org.apache.maven.artifact.resolver.DefaultArtifactResolver.resolve (DefaultArtifactResolver.java:504)
        at org.apache.maven.artifact.resolver.DefaultArtifactResolver.resolveWithExceptions (DefaultArtifactResolver.java:358)
        at org.apache.maven.artifact.resolver.DefaultArtifactResolver.resolveTransitively (DefaultArtifactResolver.java:352)
        at org.apache.maven.artifact.resolver.DefaultArtifactResolver.resolveTransitively (DefaultArtifactResolver.java:324)
        at org.apache.maven.artifact.resolver.DefaultArtifactResolver.resolveTransitively (DefaultArtifactResolver.java:287)
        at org.apache.maven.artifact.resolver.DefaultArtifactResolver.resolveTransitively (DefaultArtifactResolver.java:261)
        at org.apache.maven.artifact.resolver.DefaultArtifactResolver.resolveTransitively (DefaultArtifactResolver.java:298)
        at org.codehaus.mojo.license.api.DefaultDependenciesTool.resolve (DefaultDependenciesTool.java:497)
        at org.codehaus.mojo.license.api.DefaultDependenciesTool.loadProjectArtifacts (DefaultDependenciesTool.java:444)
        at org.codehaus.mojo.license.AbstractThirdPartyReportMojo.createThirdPartyDetails (AbstractThirdPartyReportMojo.java:555)
        at org.codehaus.mojo.license.AggregatorThirdPartyReportMojo.createThirdPartyDetails (AggregatorThirdPartyReportMojo.java:127)
        at org.codehaus.mojo.license.AbstractThirdPartyReportMojo.executeReport (AbstractThirdPartyReportMojo.java:361)

so I'm stuck.

--- a/src/it/aggregate-third-party-report/postbuild.groovy
+++ b/src/it/aggregate-third-party-report/postbuild.groovy
@@ -22,3 +22,12 @@
 
 file = new File(basedir, 'target/site/aggregate-third-party-report.html');
 assert file.exists();
+content = file.text;
+assert !content.contains('the project has no dependencies.');
+assert !content.contains('/third-party-report.html#');
+assert content.contains('<a href="#commons-logging:commons-logging:1.1.1">commons-logging:commons-logging:1.1.1</a>');
+assert content.contains('<td>The Apache Software License, Version 2.0</td>'); // TODO Should be a link
+assert content.contains('<a href="#Overview">Back to top</a>');
+
+file = new File(basedir, 'target/site/third-party-report.html');
+assert !file.exists();

--- a/src/main/java/org/codehaus/mojo/license/AbstractLicenseReportRenderer.java
+++ b/src/main/java/org/codehaus/mojo/license/AbstractLicenseReportRenderer.java
@@ -217,7 +217,7 @@ public abstract class AbstractLicenseReportRenderer
 
         sink.tableCell();
         String gav = getGAV( details );
-        sink.link( "./third-party-report.html#" + gav );
+        sink.link( "#" + gav );
         sink.text( gav );
         sink.link_();
 

--- a/src/main/java/org/codehaus/mojo/license/ThirdPartyReportRenderer.java
+++ b/src/main/java/org/codehaus/mojo/license/ThirdPartyReportRenderer.java
@@ -261,7 +261,7 @@ public class ThirdPartyReportRenderer
         sink.sectionTitle2_();
         renderThirdPartyDetailTable( detail );
 
-        sink.link( "./third-party-report.html#" + getText( "report.overview.title" ) );
+        sink.link( "#" + getText( "report.overview.title" ) );
         sink.text( getText( "report.back.to.top.page" ) );
         sink.link_();
         sink.lineBreak();

